packages/atomic/src/components/common/result-children/show-hide-button.spec.ts

```diff
@@ -0,0 +1,145 @@
+import {html} from 'lit';
+import {describe, expect, it, vi} from 'vitest';
+import {page} from 'vitest/browser';
+import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
+import {
+  renderShowHideButton,
+  type ShowHideButtonProps,
+} from './show-hide-button';
+
+describe('#renderShowHideButton', () => {
+  const defaultProps: ShowHideButtonProps = {
+    moreResultsAvailable: false,
+    loadFullCollection: vi.fn(),
+    showInitialChildren: true,
+    toggleShowInitialChildren: vi.fn(),
+    loadAllResults: 'Load all results',
+    collapseResults: 'Collapse results',
+  };
+
+  const renderComponent = async (props: Partial<ShowHideButtonProps> = {}) => {
+    return await renderFunctionFixture(
+      html`${renderShowHideButton({props: {...defaultProps, ...props}})}`
+    );
+  };
+
+  const locators = {
+    get button() {
+      return page.getByRole('button');
+    },
+  };
+
+  it('should render a button', async () => {
+    await renderComponent();
+    await expect.element(locators.button).toBeInTheDocument();
+  });
+
+  it('should apply the correct part attribute', async () => {
+    const element = await renderComponent();
+    const button = element.querySelector('button');
+    expect(button).toHaveAttribute('part', 'show-hide-button');
+  });
+
+  it('should apply the correct class', async () => {
+    const element = await renderComponent();
+    const button = element.querySelector('button');
+    expect(button).toHaveClass('show-hide-button');
+  });
+
+  describe('when showInitialChildren is true', () => {
```

Comment:
Nitpick, we can combine the condition with the it here:

"should display loadAllResults text when showInitialChildren is true"

Reply:
There are similar occurrences in the spec; if you apply the comment, apply it everywhere.